packages/paste-core/layout/stack/src/index.tsx

```diff
@@ -0,0 +1,122 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {useUID} from 'react-uid';
+import {FlexboxProps, LayoutProps, MarginProps, ResponsiveValue} from 'styled-system';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import {Display, Space, isSpaceTokenProp, ResponsiveProp} from '@twilio-paste/style-props';
+
+export type StackOrientationOptions = 'horizontal' | 'vertical';
+export type StackOrientation = ResponsiveValue<StackOrientationOptions>;
+
+export interface StackProps {
+  orientation?: StackOrientation;
+  spacing?: Space;
+}
+
+interface StackStyleProps extends LayoutProps, FlexboxProps {}
+
+const getDisplay = (orientation: StackProps): Display => {
+  if (Array.isArray(orientation)) {
+    return (orientation as StackOrientationOptions[]).map((value: StackOrientationOptions) => {
+      if (value === 'horizontal') {
+        return 'flex';
+      }
+
+      return 'block';
+    });
+  }
+
+  if (orientation === 'horizontal') {
+    return 'flex';
+  }
+
+  return 'block';
+};
+
+const getStackStyles = (orientation: StackProps): StackStyleProps => {
+  const styles: StackStyleProps = {
+    display: getDisplay(orientation),
+    alignItems: 'center',
+    flexWrap: 'wrap',
+  };
+
+  return styles;
+};
+
+const getChildMargins = (orientation: StackOrientation, spacing: Space): MarginProps => {
+  const styles: MarginProps = {};
+
+  if (Array.isArray(orientation)) {
+    const horizontalMarginArray: Space = [];
+    const verticalMarginArray: Space = [];
+
+    orientation.forEach((value, i) => {
+      let marginRightValue;
+      let marginBottomValue;
+
+      if (value === 'horizontal') {
+        marginRightValue = spacing;
+        marginBottomValue = 'space0';
+      }
+      if (value === 'vertical') {
+        marginRightValue = 'space0';
+        marginBottomValue = spacing;
+      }
+
+      // @ts-ignore
+      horizontalMarginArray[i] = marginRightValue;
+      // @ts-ignore
+      verticalMarginArray[i] = marginBottomValue;
```

Comment:
I believe you might be able to simplify this to something like

```
horizontalMarginArray[i] = (value=== 'horizontal') ? spacing : 'space0'

verticalMarginArray[i] = (value=== 'horizontal') ? 'space0' : spacing;
```

Reply:
In fact could you further simply this entire thing to 
```
const styles: MarginProps = {}

if (Array.isArray(orientation)) {
  const marginRight: Space = [];
  const marginBottom: Space = [];

  orientation.forEach((value, i) => {
    marginRight[i] = (value=== 'horizontal') ? spacing : 'space0'
    marginBottom[i] = (value=== 'horizontal') ? 'space0' : spacing;
  });

  styles = {
    marginRight,
    marginBottom
  }
}

  if (orientation === 'horizontal') {
    styles.marginRight = spacing;
  }

  if (orientation === 'vertical') {
    styles.marginBottom = spacing;
  }

  return styles;
```

Reply:
That is much simpler. Just needed to change `const styles...` to `let styles...`. Thanks.